proposed/2023/release-staging-and-deprecation.md

```diff
@@ -0,0 +1,90 @@
+# ***Release staging and deprecation***
+<!-- Replace `Title` with an appropriate title for your design -->
+
+- Author Name Artak Mkrtchyan ([mkArtakMSFT](https://github.com/mkArtakMSFT))
+- GitHub Issue <https://github.com/NuGet/NuGetGallery/issues/3931>
+
+## Summary
+
+This proposal introduces a concept of a logical grouping for packages published to NuGet.
+This will allow one to easily identify all the packages released as part of a single product version release as well as request bulk-processing for a set of packages which belong to the same release.
+For example, all packages published as part of .NET 8 Preview 5 release will be staged together and then also can be deprecated together by a single call to NuGet APIs.
+
+## Motivation 
+
+<!-- Why are we doing this? What pain points does this solve? What is the expected outcome? -->
+Many products (Apps / Tools / Frameworks / Utilities / etc. ...) consist of multiple interdependent packages which should be published together as a single set. Currently, NuGet provides no functionality for definding / grouping packages into such sets. This results in package owners having to keep track of these package sets on their own. There are three specific sets of scenarios which this feature will simplify dramatically:
+1. When preparing a release, product owners would like to have a mechanism to pre-stage their product packages, so that the release will consist only of a single action of marking the set of packages as `published`, which would result all the packages in the set to be marked as published at the same time.
+2. Sometimes developers have to take a dependency on several related packages in their projects, which are part of the same "release". Without having a concept of the `release` developers have to guess which versions of those packages are compatible. As a result, most of hte time developers would simply pick the latest version of all these packages, which somtimes can lead to incompatibility issues down the road.
+3. Keeping a package healthy on NuGet also means keeping the list of packages aligned with it support policy. Hence, package owners try to deprecate packages which are out of support. Without ability to group packages, package owners end up keeping track of all the packages which belong to the same release and run the deprecation over a discrete set of packages when necessary. Being able to group packages together would enable deprecating a whole release at once, knowing that all the packages which belong to that release will get deprecated.
+
+## Explanation
+
+### Functional explanation
```

Comment:
Should symbol packages (.snupkgs) be stage-able also? If so, and we opt to use a release label that's in the .nuspec, then .snupkg pack would need to be enhanced also. I believe the .nuspec in the .snupkg is some subset of the related .nupkg .nuspec.

Reply:
Given that the "group" information is no longer in the package metadata, do we have to treat symbols packages differently?

Reply:
Sorry, missed this response. No, they would not be different then. No metadata change but `nuget push` probably needs a gesture to "push to stage" and this would work on .nupkg or .snupkg.